library/worldgen/material_rule/src/testmod/java/org/quiltmc/qsl/worldgen/material_rule/QuiltMaterialRuleTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.worldgen.material_rule;
+
+import net.minecraft.block.Blocks;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryKey;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.BiomeKeys;
+import net.minecraft.world.gen.noise.NoiseParametersKeys;
+import net.minecraft.world.gen.surfacebuilder.SurfaceRules;
+
+import org.quiltmc.loader.api.ModContainer;
+import org.quiltmc.qsl.base.api.entrypoint.ModInitializer;
+import org.quiltmc.qsl.worldgen.material_rule.api.SurfaceMaterialRuleContext;
+import org.quiltmc.qsl.worldgen.material_rule.api.SurfaceMaterialRuleEvents;
+
+public class QuiltMaterialRuleTest implements SurfaceMaterialRuleEvents.OverworldModifierCallback,
+		SurfaceMaterialRuleEvents.NetherModifierCallback,
+		SurfaceMaterialRuleEvents.TheEndModifierCallback {
+	@Override
+	public void modifyOverworldRules(SurfaceMaterialRuleContext.Overworld context) {
+		//when in doubt, T R A N S. seed 7205143747332514273 is a good one for testing.
```

Comment:
Nitpick to a comment that's already pretty informal :P
```suggestion
		// When in doubt, T R A N S. Seed 7205143747332514273 is a good one for testing.
```